admiral/camp_functions.sqf

```diff
@@ -275,73 +271,79 @@ adm_camp_fnc_getSpawnFunction = {
-        _spawnedGroups = [];
-        _spawnedGroups = [_zone, GROUP_TYPE_TECH, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnTechGroups] call adm_camp_fnc_trySpawnGroups;
-        _zoneTechGroups append _spawnedGroups;
-        INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' technical group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
-        _spawnedGroups = [];
-        _spawnedGroups = [_zone, GROUP_TYPE_ARMOUR, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnArmourGroups] call adm_camp_fnc_trySpawnGroups;
-        _zoneArmourGroups append _spawnedGroups;
-        INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' armour group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
-        ["camp.spawned.groups", [_zoneInfGroups, _zoneTechGroups, _zoneArmourGroups, _zone]] call adm_event_fnc_emitEvent;
-
-        sleep GET_CAMP_DELAY(_zone);
-        [_zone] call adm_camp_fnc_isPoolEmpty || {!IS_CAMP_ENABLED(_zone)};
-    };
-    INFO("admiral.camp",FMT_1("Periodic Camp Zone '%1' has stopped.",GET_ZONE_ID(_zone)));
+
+    [
+        {
+            params ["_args", "_id"];
+            _args params ["_zone","_zoneInfGroups","_zoneTechGroups","_zoneArmourGroups"];
+
+            if ([_zone] call adm_camp_fnc_isPoolEmpty || {!IS_CAMP_ENABLED(_zone)}) exitWith {
+                INFO("admiral.camp",FMT_1("Periodic Camp Zone '%1' has stopped.",GET_ZONE_ID(_zone)));
+                _id call CBA_fnc_removePerFrameHandler;
+            };
+
+            private _spawnedGroups = [];
+            _spawnedGroups = [_zone, GROUP_TYPE_INF, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnInfGroups] call adm_camp_fnc_trySpawnGroups;
+            _zoneInfGroups append _spawnedGroups;
+            INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' infantry group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
+            _spawnedGroups = [];
+            _spawnedGroups = [_zone, GROUP_TYPE_TECH, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnTechGroups] call adm_camp_fnc_trySpawnGroups;
+            _zoneTechGroups append _spawnedGroups;
+            INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' technical group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
+            _spawnedGroups = [];
+            _spawnedGroups = [_zone, GROUP_TYPE_ARMOUR, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnArmourGroups] call adm_camp_fnc_trySpawnGroups;
+            _zoneArmourGroups append _spawnedGroups;
+            INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' armour group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
+            ["camp.spawned.groups", [_zoneInfGroups, _zoneTechGroups, _zoneArmourGroups, _zone]] call adm_event_fnc_emitEvent;
+        },
+        GET_CAMP_DELAY(_zone),
```

Comment:
FYI, I just realized that by using the per frame handler, the delay expression is evaluated only once, when the handler is added. I'm not sure if this affects anything at the moment, because as far as I know it doesn't change during a mission, but we should be aware of this.

Reply:
This is correct, the PFH delay variable is cached once on loop and kept forever. I'm not aware of anyone ever dynamically changing this on-the-fly so hopefully won't make any difference.